storage/innobase/row/row0merge.cc

```diff
@@ -5197,8 +5212,17 @@ dberr_t row_merge_bulk_t::bulk_insert_buffered(const dtuple_t &row,
     if (buf->n_tuples == 0)
     {
       /* Tuple data size is greater than srv_sort_buf_size */
-      dtuple_t *big_tuple= row_merge_buf_large_tuple(
-        row, &m_blob_file, &large_tuple_heap);
+      dtuple_t *big_tuple;
+      if (!large_tuple_heap)
+        large_tuple_heap= mem_heap_create(DTUPLE_EST_ALLOC(row.n_fields));
+
+      big_tuple= dtuple_copy(&row, large_tuple_heap);
```

Comment:
We seem to be missing `ut_ad(i == 0)` here. There can’t be any BLOBs in other indexes than the clustered index, right?

I’d declare and initialize `big_tuple` in a single statement.

At the start of this function, can we assert `ut_ad(row.n_fields == ind.n_fields)`?